sdk/logical/rotation_job.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package logical
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/robfig/cron/v3"
+)
+
+// RotationOptions is an embeddable struct to capture common lease
+// settings between a Secret and Auth
+type RotationOptions struct {
+	// Schedule holds the info for the framework.Schedule
+	Schedule *RootSchedule
+}
+
+// RotationJob represents the secret part of a response.
+type RotationJob struct {
+	RotationOptions
+
+	// RotationID is the ID returned to the user to manage this secret.
+	// This is generated by Vault core. Any set value will be ignored.
+	// For requests, this will always be blank.
+	RotationID string `sentinel:""`
+	Path       string
+	Name       string
+}
+
+type RotationJobConfigureRequest struct {
+	Name             string
+	MountPoint       string
+	ReqPath          string
+	RotationSchedule string
+	RotationWindow   int
+	RotationTTL      int
+}
+
+func (s *RotationJob) Validate() error {
+	// TODO: validation?
+	return nil
+}
+
+func getRotationJob(rotationSchedule, path, rotationJobName string, rotationWindow, ttl int) (*RotationJob, error) {
```

Comment:
Should this be named something like newRotationJob or makeRotationJob? The Get part of the name makes me think it should be doing a read.